src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingNullCaseInSwitchCheck.java

```diff
@@ -0,0 +1,201 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2024 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.coding;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
+
+/**
+ * <p>
+ * Checks that a given switch statement or expression that use a reference type in its selector
```

Comment:
please see the update I added another section below to explain what we validate specifically

Reply:
Let's rearrange this a bit more if you agree.

Description
Rationale
Validates
Limitations

IMO, this is a more natural flow. I would also update limitations with a bit more verbiage before "Limitations:". Something like we know there are more cases where there are reference types, but we can't identify them because it requires type knowledge.

> This check does not support enums

Isn't there another option where we can't tell the difference between primitive and their class versions? Are there any other limitations not really being mentioned?

Reply:
Maybe we don't need a fully documented limitations section. We define what we validate. I am not seeing how that doesn't cover the limitations fully. Why not just trim limitations to a simple sentence, no list. Let me know if you agree.

Reply:
I tend to agree the list was suitable when we don't specifically mention what we support we just say reference type then we exclude some items in the limitation now we have mentioned what we validate exactly, there is no reason for a separate limitation list I just will leave one sentence (not a list) for enums to be super clear that we don't support it

Reply:
> Description
> Rationale
> Validates
> Limitations

> Maybe we don't need a fully documented limitations section. We define what we validate. I am not seeing how that doesn't cover the limitations fully. Why not just trim limitations to a simple sentence (or 2), no list. Let me know if you agree.

done